samples/spookyshots/main.py

```diff
@@ -0,0 +1,139 @@
+    
+    This app leverages Cloudinary's powerful Generative AI features to make your pets look extra spooky this Halloween. Try it out, and share the spooky transformations with your friends!
+    """)
+
+if selected == "Spooky Pet Background Generator":
+    st.title("Spooky Halloween Pet Image Transformer")
+
+    uploaded_file = st.file_uploader("Upload an image (jpg, jpeg, png)", type=["jpg", "jpeg", "png"])
+
+    default_prompt = "A dark foggy Halloween night with a full moon in the sky surrounded by twisted trees Scattered glowing pumpkins with carved faces placed around an old broken fence in the background a shadowy haunted house with dimly lit windows"
+
+    modify_prompt = st.checkbox("Do you want to modify the generative Halloween background prompt?", value=False)
+
+    custom_prompt = st.text_input(
+        "Optional: Modify the generative Halloween background prompt",
+        value=default_prompt,
+        disabled=not modify_prompt
+    )
+
+    if st.button("Submit"):
+        if uploaded_file:
+            if uploaded_file.size > MAX_FILE_SIZE_BYTES:
+                st.warning(f"File size exceeds the 5 MB limit. Please upload a smaller file.")
+            else:
+                message_placeholder = st.empty()
+                message_placeholder.write("Generating image... Please have patience while the image is being processed by Cloudinary.")
+                
+                upload_result = cloudinary.uploader.upload(
+                    uploaded_file, 
+                    public_id=f"user_uploaded_{uploaded_file.name[:6]}", 
+                    unique_filename=True, 
+                    overwrite=False
+                )
+                public_id = upload_result['public_id']
+                halloween_bg_image_url = CloudinaryImage(public_id).image(
+                    effect=f"gen_background_replace:prompt_{custom_prompt}"
+                )
+                
+                start_index = halloween_bg_image_url.find('src="') + len('src="')
+                end_index = halloween_bg_image_url.find('"', start_index)
+                image_url = halloween_bg_image_url[start_index:end_index] if start_index != -1 and end_index != -1 else None
+                
+                if image_url:
+                    message_placeholder.empty()
+                    st.image(image_url)
+                else:
+                    message_placeholder.write("Failed to apply the background. Please try again.")
+        else:
+            st.write("Please upload an image to proceed.")
+
+    st.write("If you don't see the image, just refresh the page and try again. You can also use this test image: [Test Image](https://unsplash.com/photos/black-and-white-cat-lying-on-brown-bamboo-chair-inside-room-gKXKBY-C-Dk)")
```

Comment:
Nice to have, but you can skip it:

1. Putting a spinning wheel would greatly improve user experience.
2. Cloudinary allows uploading images from URLs (instead of file you can just pass a URL). Instead of going to that website, downloading the image and re-uploading it.

Reply:
Add url support and spinner functionality